jsonschema-use-cases.xml

```diff
@@ -0,0 +1,310 @@
+                <t>
+                    For example, if a value is specified to be a date, the form field where the value is specified can provide immediate feedback if an invalid date is entered, and even offer a date picker to help the user input a correct value.
+                </t>
+                <t>
+                    Weak interoperability requirements can hamper the user experience within this use case. If the schema is ambiguous in any way, or is up to the discretion of the customer's user-agent, some customers may have a difficult time submitting a correct request.
+                </t>
+            </section>
+
+            <section title="Fuzzing, enumeration, and generation">
+                <t>
+                    Security applications need to generate examples of JSON documents within the valid set, and outside the valid set.
+                </t>
+            </section>
+
+            <section title="Embedded database constraints">
+                <t>
+                    A database that uses JSON may need a way to declare, enforce, or guarantee certain constraints on the JSON document being stored. The database may also use JSON Schema as a way to annotate certain fields as having a special meaning for uniqueness or indexing purposes.
+                </t>
+            </section>
+
+            <section title="Partial validation" anchor="partial-validation">
+                <t>
+                    Due to technical limitations, some JSON parsers may only be able to understand a subset of the JSON value space, and it makes sense to validate the value read by the application, instead of the JSON document provided to the JSON parser. For example:
+                </t>
+                <list style="symbols">
+                    <t>Many JSON parsers cast the arbitrary-precision decimal numbers to an IEEE floating point, validating the number after it has lost some precision.</t>
+                    <t>Some programming languages cannot distinguish between an ordered array of values and a key-value map; or an empty array is identical to an empty object.</t>
+                    <t>Other validators may have a limited amount of memory and cannot support assertions more complicated than a deterministic context-free grammar.</t>
+                </list>
+                <t>
+                    Users of these validators need a way to determine if the missing functionality is essential to correctly validating input, and if not, get a validation result that would be correct but-for the unimplemented functionality.
+                </t>
+                <t>
+                    <cref>
+                        This should work both ways; an application should be able to use a third-party schema and understand if the assertions go beyond what the environment supports; and an application should be able to publish a schema that is compatible with the subset of the value space that it supports. Consider if these are separate use cases.
```

Comment:
> Consider if these are separate use cases.

I'm not sure what the intent behind this sentence is.  Is it an exercise for the reader?